scene/resources/texture.cpp

```diff
@@ -3547,3 +3547,82 @@ PlaceholderTextureLayered::~PlaceholderTextureLayered() {
 	ERR_FAIL_NULL(RenderingServer::get_singleton());
 	RS::get_singleton()->free(rid);
 }
+
+/////////////////////////////////////////////////
+
+void Texture2DRD::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("set_texture_rd_rid", "texture_rd_rid"), &Texture2DRD::set_texture_rd_rid);
+	ClassDB::bind_method(D_METHOD("get_texture_rd_rid"), &Texture2DRD::get_texture_rd_rid);
+
+	ADD_PROPERTY(PropertyInfo(Variant::RID, "texture_rd_rid"), "set_texture_rd_rid", "get_texture_rd_rid");
+}
+
+int Texture2DRD::get_width() const {
+	return size.width;
+}
+
+int Texture2DRD::get_height() const {
+	return size.height;
+}
+
+RID Texture2DRD::get_rid() const {
+	return texture_rid;
+}
+
+bool Texture2DRD::has_alpha() const {
+	return false;
```

Comment:
How important is reporting this correctly? The only way I can think of doing this is checking each and every `DATA_FORMAT_...` and seeing which ones have an alpha channel. I guess we could add a function for this to `RD::TextureFormat`.